### PR TITLE
Allow read-only Cypher functions in cypher_raw

### PR DESCRIPTION
**Pull request by JeroenDeDauw**
Builds on #471 and kept separate for easy review. "Files changed" tab correctly only shows the extra stuff

Reviewed by Jeroen and Claude

The CypherQueryFilter was blocking all function calls, but built-in Cypher functions like `type()`, `labels()`, `count()` are read-only and safe.

## Changes

- Allow read-only function calls (`type`, `labels`, `count`, etc.)
- Block `CALL` (stored procedures can have side effects)
- Block `LOAD` (file system access via LOAD CSV)
- Add unicode escape normalization to prevent obfuscation attacks like `\u000DCREATE`
- Update demo page to show relation types using `type(r)`
- Add comprehensive security tests (55 test cases covering standard writes, case variations, procedure calls, file access, unicode obfuscation, subquery attacks, and more)

## Security

The filter now blocks dangerous operations while allowing safe read-only functions:

| Blocked | Reason |
|---------|--------|
| CREATE, SET, DELETE, REMOVE, MERGE, DROP | Write operations |
| CALL | Stored procedures can have side effects |
| LOAD | File system access via LOAD CSV |

All 55 security test cases pass, covering obfuscation attempts like unicode escapes, comment hiding, subqueries, and case variations.

### CODE REVIEW

File: src/CypherQueryFilter.php [hunk 1]
```diff
@@ -7,21 +7,14 @@
 class CypherQueryFilter {
 
 	private const array WRITE_KEYWORDS = [
-		'CREATE', 'SET', 'DELETE', 'REMOVE', 'MERGE', 'DROP'
+		'CREATE', 'SET', 'DELETE', 'REMOVE', 'MERGE', 'DROP',
+		'CALL', 'LOAD', 'FOREACH'
```
[the remark this answers is not in the document]
Comment (JeroenDeDauw) [reply src/CypherQueryFilter.php]: @copilot where in the codebase do we use SHOW? Are those queries going through this CypherQueryFilter service?

[the remark this answers is not in the document]
Comment (malberts) [reply src/CypherQueryFilter.php]: I think the rest of the point is still valid. A user can get information that likely should not be public:

Image
